### DISCUSSION TITLE
멀티플레이 동기화 관련 질문입니다 :(

### DISCUSSION THREAD

**Discussion by kimyang-sun** [naverz/zepeto-studio-kor#984] [5. Q&A]
멀티플레이 동기화 관련해서 궁금한점이 있습니다 :)
멀티플레이 관련한 내용을 찾아봤는데 동기화라는 개념 자체가 잘 안잡혀서요 ㅠ

멀티플레이 예제를 참고하여 기본 움직임이나 state값(점프 등)에 대한 동기화는 기본적으로 구현이 되었는데
캐릭터가 순간이동을 할때에는 사라져버려서 따로 동기화를 추가해줘야 하는건가 싶어서 여쭤봅니다.

Teleport 내장함수를 사용하지 않구 아래 이미지처럼 enabled 로 없애주고 해당하는 오브젝트의 좌표를 받아와 적용시켜주고
다시 enabled를 true로 해주었습니다.
position 값에 대한 동기화는 이미 적용이 된 상태라고 판단해서 동기화 코드를 따로 추가안해줬는데 멀티플레이 테스트 링크로 테스트를 하니까 캐릭터가 해당 포지션으로 가려고 하는거 같긴한데 enabled가 안먹고 그냥 목적지의 위치로 뒤돌아서 뛰어가네요 :(
enabled 에 대한 동기화를 안해줘서 순간이동하듯이 안가고 해당 포지션값으로 그냥 뛰어가는걸까요??

![image](https://user-images.githubusercontent.com/56675004/178201373-4b2c8268-f27d-4c3c-a7fe-2e63e5ea5c05.png)

추가로 동기화에 대한 이해를 제대로 못하고 있는 상태라 어느 부분이 동기화가 필요한지 개념이 잘 안잡히네요 ㅠ..
어떤 기능을 추가하든 전부다 동기화를 해줘야 한다고 생각하면 편할까요?.?
+ 예를 들어 npc같이 오브젝트가 특정위치를 돌아다닌다고 하면 모든 유저에게 npc위치가 동일하게 보이려면 npc의 위치도 따로 동기화를 해줘야 하는 부분
+ 열차를 탑승하는 기능을 구현했을때 열차에 탑승해있는 캐릭터의 앉는 모션의 애니메이션, 열차의 위치, 탑승인원수를 초과하면 타기 버튼을 막는 기능 등등

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 모두에게 영향을 주는 정보는 서버 코드로 작성하고 내 단말기에서만 보여지는 이벤트는 클라이언트로 구현한다고 개념을 잡으시면 좋을 것 같습니다.

탑승인원처럼 모두가 알아야 하는 정보는 서버코드에서 변수로 관리하고,
클라이언트는 탑승을 하려고 할때 탑승인원 정보를 서버로부터 받아온 후, 초과면 버튼 클릭이 불가하고 아니면 클릭이 가능하게 하도록 버튼 이벤트를 구현하면 될 것입니다.

NPC의 위치가 모두에게 동기화가 되어야 하는 중요한 요소일 경우(예: 보스몹) 동기화를 시켜주어야 합니다.
반면 단순히 배경에 돌아다니는 새처럼 단말기마다 따로 보여줘도 되는 NPC는 동기화를 해주지 않아도 됩니다.

**Reply by kimyang-sun** [reply]
말씀해주신 흐름으로 동기화 적용 해보겠습니다!
답변 감사합니다 :)
